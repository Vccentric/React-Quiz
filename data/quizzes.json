{
  "quizzes": [
    {
      "name": "Beginner Mathematics",
      "questions": [
        {
          "question": "How many sides are there in a pentagon?",
          "answerChoices": ["4", "5", "6", "7"],
          "correctAnswer": "5"
        },
        {
          "question": "What is: 16 x 16 = ?",
          "answerChoices": ["192", "252", "256", "264"],
          "correctAnswer": "256"
        },
        {
          "question": "How many sides are there in a octagon?",
          "answerChoices": ["5", "6", "7", "8"],
          "correctAnswer": "8"
        },
        {
          "question": "What is: 361 / 19 = ?",
          "answerChoices": ["19", "17", "20.056", "18.05"],
          "correctAnswer": "19"
        },
        {
          "question": "How many sides are there in a star?",
          "answerChoices": ["10", "11", "12", "14"],
          "correctAnswer": "10"
        }
      ]
    },
    {
      "name": "Advanced Mathematics",
      "questions": [
        {
          "question": "What is: 21 x 17 = ?",
          "answerChoices": ["273", "331", "336", "357"],
          "correctAnswer": "357"
        },
        {
          "question": "How many sides are there in a heptagon?",
          "answerChoices": ["3", "5", "7", "9"],
          "correctAnswer": "7"
        },
        {
          "question": "What is: 84 / 4 = ?",
          "answerChoices": ["16", "18", "21", "24"],
          "correctAnswer": "21"
        },
        {
          "question": "How many sides are there in a triange?",
          "answerChoices": ["3", "4", "5", "6"],
          "correctAnswer": "3"
        },
        {
          "question": "What is: 11 x 7 = ?",
          "answerChoices": ["71", "77", "78", "79"],
          "correctAnswer": "77"
        }
      ]
    }
  ]
}
