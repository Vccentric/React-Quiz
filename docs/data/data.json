{
    "questions": [
        {
            "question": "How many sides are there in a pentagon?",
            "answerOptions": [
                "4",
                "5",
                "6",
                "7"
            ],
            "answerIndex": 1
        },
        {
            "question": "What is: 16 x 16 = ?",
            "answerOptions": [
                "192",
                "252",
                "256",
                "264"
            ],
            "answerIndex": 2
        },
        {
            "question": "How many sides are there in a octagon?",
            "answerOptions": [
                "5",
                "6",
                "7",
                "8"
            ],
            "answerIndex": 3
        },
        {
            "question": "What is: 361 / 19 = ?",
            "answerOptions": [
                "19",
                "17",
                "20.056",
                "18.05"
            ],
            "answerIndex": 0
        },
        {
            "question": "How many sides are there in a star?",
            "answerOptions": [
                "10",
                "11",
                "12",
                "14"
            ],
            "answerIndex": 0
        },
        {
            "question": "What is: 21 x 17 = ?",
            "answerOptions": [
                "273",
                "331",
                "336",
                "357"
            ],
            "answerIndex": 3
        },
        {
            "question": "How many sides are there in a heptagon?",
            "answerOptions": [
                "3",
                "5",
                "7",
                "9"
            ],
            "answerIndex": 2
        },
        {
            "question": "What is: 84 / 4 = ?",
            "answerOptions": [
                "16",
                "18",
                "21",
                "24"
            ],
            "answerIndex": 2
        },
        {
            "question": "How many sides are there in a triange?",
            "answerOptions": [
                "3",
                "4",
                "5",
                "6"
            ],
            "answerIndex": 0
        },
        {
            "question": "What is: 11 x 7 = ?",
            "answerOptions": [
                "71",
                "77",
                "78",
                "79"
            ],
            "answerIndex": 1
        }
    ]
}