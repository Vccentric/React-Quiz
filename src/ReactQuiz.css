/**
 * React Quiz v1.0.0
 * CSS file for the react quiz application
 *
 * @author      Christopher Viray
 * @copyright   (c) 2018 Christopher Viray
 * @license     MIT
 * @version     1.0.0
 */

body {
    margin: 0;
    padding: 0;
    font-size: 20px;
}

#root {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    height: 100vh;
}

.popup {
    width: 80%;
    max-width: 700px;
    border: 1px solid gray;
    text-align: center;
    box-shadow: 2px 2px 2px 0 gray;
}

.popup h1 {
    font-weight: bold;
    margin: 0;
    padding: 10px 0;
    background-color: lightskyblue;
}

button {
    cursor: pointer;
    margin: 10px auto;
}

#quiz {
    width: 80%;
    max-width: 700px;
}

#quizInfo {
    display: flex;
    flex-direction: column;
    border: 1px solid gray;
    box-shadow: 2px 2px 2px 0 gray;
}

#quizStatus {
    display: flex;
    padding: 10px;
    background-color: lightskyblue;
}

#questionCount {
    flex-grow: 1;
}

#score {
    flex-grow: 1;
    text-align: right;
}

#question {
    margin: 0;
    padding: 20px;
}

#answerOptions {
    list-style: none outside none;
    margin: 0;
    padding: 0;
}

.answer {
    display: flex;
    cursor: pointer;
    border: 1px solid gray;
    background-color: white;
    margin: 15px 0;
    box-shadow: 2px 2px 2px 0 gray;
}

.answer:hover {
    background-color: whitesmoke;
}

.answer span {
    padding: 10px;
}

.answer p {
    margin: 0;
    padding: 10px;
}

.answer.disabled {
    cursor: default;
    background-color: white;
}

.answer.correct {
    cursor: default;
    background-color: #b1ffb1;
}

.answer.wrong {
    cursor: default;
    background-color: #ffc4c4;
}

#message {
    font-weight: bold;
    text-align: center;
    margin: 5px auto;
    color: black;
}

#message.correct {
    color: green;
}

#message.wrong {
    color: red;
}

#quizControls {
    visibility: visible;
    border: 1px solid gray;
    text-align: center;
    box-shadow: 2px 2px 2px 0 gray;
}

#quizControls.hidden {
    visibility: hidden;
}
