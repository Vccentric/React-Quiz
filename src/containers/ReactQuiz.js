import React, { useState } from "react";
import PropTypes from "prop-types";
import QuizList from "../components/QuizList";
import Quiz from "./Quiz";

// main container component
const ReactQuiz = ({ data }) => {
  // setup state
  const [view, setView] = useState({
    showQuizList: true,
    showQuiz: false,
  });

  return (
    <div id="react-quiz">
      {view?.showQuizList && <QuizList data={data} setView={setView} />}
      {view?.showQuiz && <Quiz {...view.showQuiz} setView={setView} />}
    </div>
  );
};

ReactQuiz.propTypes = {
  data: PropTypes.object,
};

export default ReactQuiz;
