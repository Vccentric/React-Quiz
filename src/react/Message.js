import React from 'react';
import { hot } from 'react-hot-loader';

// message component
function Message(props) {
    const { isAnswered, isCorrect } = props;
    return (
        <div
            id="message"
            style={isAnswered ? { display: 'block' } : { display: 'none' }}
            className={isCorrect ? 'correct' : 'wrong'}
        >
            {isCorrect ? 'Correct Answer!' : 'Wrong Answer!'}
        </div>
    );
}

export default hot(module)(Message);
