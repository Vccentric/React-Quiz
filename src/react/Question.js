import React from 'react';
import { hot } from 'react-hot-loader';

// question component
function Question(props) {
    const { question } = props;
    return <p id="question">{question}</p>;
}

export default hot(module)(Question);
