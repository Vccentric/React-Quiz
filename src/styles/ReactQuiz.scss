#root {
  align-items: center;
  background-color: #e9e9e9;
  display: flex;
  height: 100vh;
  justify-content: center;
  width: 100%;
}

#react-quiz {
  border: 10px solid white;
  border-radius: 10px;
  font-size: 16px;
  flex-basis: 100%;
  max-width: 500px;
  .header {
    background-color: #8a80b5;
    color: white;
    text-align: center;
    .title {
      font-size: 2rem;
      margin: 1rem 0;
    }
  }
  .body {
    background-color: #f9f9f9;
    padding: 2rem;
  }
  ul {
    list-style-type: none;
    margin: 0;
    padding: 0;
  }
  button {
    all: unset;
    background-color: white;
    border: 1px solid gray;
    border-radius: 5px;
    box-sizing: border-box;
    color: black;
    cursor: pointer;
    font-size: 1.4rem;
    font-weight: bold;
    margin: 1rem 0;
    padding: 1rem;
    text-align: center;
    width: 100%;
  }
  button:hover,
  button.selected {
    background-color: #8a80b5;
    color: white;
    text-decoration: underline;
  }
  button.action {
    background-color: #8a80b5;
    color: white;
    flex-basis: 48%;
  }
  button.action:hover,
  button.selected:hover {
    background-color: #b7b0d1;
  }
  button.action:disabled {
    background-color: gray;
    color: white;
    cursor: default;
  }
  button.action:disabled:hover {
    text-decoration: none;
  }
  .actions-container {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
  }
}

#quiz-list {
  display: flex;
  flex-direction: column;
  .body {
    .title {
      margin: 1rem 0;
    }
    .quiz {
      background-color: white;
      border: 1px solid gray;
      border-radius: 5px;
      cursor: pointer;
      margin: 1rem 0;
      padding: 1rem;
      .name {
        font-size: 1.5rem;
        margin-bottom: 0.5rem;
      }
      .total {
        color: #c1c4cd;
      }
    }
    .quiz:hover {
      border-color: #8a80b5;
      color: #8a80b5;
    }
  }
}

#quiz {
  display: flex;
  flex-direction: column;
  .body {
    .question {
      display: flex;
      flex-direction: column;
      .index-total {
        margin: 0.5rem 0;
      }
    }
  }
}
