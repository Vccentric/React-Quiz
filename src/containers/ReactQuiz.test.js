import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react";
import React from "react";
import ReactQuiz from "./ReactQuiz";
import data from "../../data/quizzes.json";

test("renders correctly - snapshot", () => {
  const tree = renderer.create(<ReactQuiz data={data} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("renders quiz list", () => {
  const initState = {
    quizList: true,
    selectedQuiz: false,
  };
  render(<ReactQuiz data={data} localState={initState} />);
  const element = screen.getByText("Welcome!!!");
  expect(element).toBeInTheDocument();
});

test("renders specific quiz", () => {
  const initState = {
    quizList: false,
    selectedQuiz: data?.quizzes?.[0],
  };
  render(<ReactQuiz data={data} localState={initState} />);
  const element = screen.getByText("Beginner Mathematics");
  expect(element).toBeInTheDocument();
});
