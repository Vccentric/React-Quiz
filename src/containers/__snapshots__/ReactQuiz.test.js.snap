// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly - snapshot 1`] = `
<div
  id="react-quiz"
>
  <div
    id="quiz-list"
  >
    <div
      className="header"
    >
      <h1
        className="title"
      >
        Welcome!!!
      </h1>
    </div>
    <div
      className="body"
    >
      <h2
        className="title"
      >
        Quizzes:
      </h2>
      <ul>
        <li>
          <div
            className="quiz"
            onClick={[Function]}
          >
            <div
              className="name"
            >
              Beginner Mathematics
            </div>
            <div
              className="total"
            >
              5
               Questions
            </div>
          </div>
        </li>
        <li>
          <div
            className="quiz"
            onClick={[Function]}
          >
            <div
              className="name"
            >
              Advanced Mathematics
            </div>
            <div
              className="total"
            >
              5
               Questions
            </div>
          </div>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
